src/hugr/views.rs

```diff
@@ -303,6 +305,29 @@ pub trait HugrView: sealed::HugrInternals {
     }
 }
 
+/// A view of the whole Hugr.
+/// (Just provides static checking of the type of the root node)
+pub struct WholeHugrView<'a, Root = Node>(&'a Hugr, PhantomData<Root>);
+
+impl<'a, Root: NodeHandle> WholeHugrView<'a, Root> {
+    /// Create a hierarchical view of a whole HUGR
+    ///
+    /// # Errors
+    /// Returns [`HugrError::InvalidNode`] if the root isn't a node of the required [OpTag]
```

Comment:
I feel we could use a `HugrError::InvalidTag` variant for these cases. This could be another PR, since we do the same on all the views.

Reply:
Yes, #581, I think it'd be good to do that first, and then I'll make this PR update those tests